tests/testsuite/tree.rs

```diff
@@ -1846,6 +1846,61 @@ c v1.0.0
         .run();
 }
 
+#[cargo_test]
+fn depth_workspace() {
+    Package::new("somedep", "1.0.0").publish();
+    Package::new("otherdep", "1.0.0").publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [workspace]
+            members = ["a", "b", "c"]
+            "#,
+        )
+        .file("a/Cargo.toml", &basic_manifest("a", "1.0.0"))
+        .file("a/src/lib.rs", "")
+        .file(
+            "b/Cargo.toml",
+            r#"
+            [package]
+            name = "b"
+            version = "0.1.0"
+
+            [dependencies]
+            c = { path = "../c" }
+            somedep = "1"
+            "#,
+        )
+        .file("b/src/lib.rs", "")
+        .file(
+            "c/Cargo.toml",
+            r#"
+            [package]
+            name = "c"
+            version = "0.1.0"
+
+            [dependencies]
+            somedep = "1"
+            otherdep = "1"
+            "#,
+        )
+        .file("c/src/lib.rs", "")
+        .build();
+
+    p.cargo("tree --depth workspace")
```

Comment:
Could you split this PR into two commits
1. Add the test without the `--depth workspace` flag
2. Add support for `--depth workspace` and update the test to pass it in

With this, it'll make the intent of this change a lot clearer and help test the test

Reply:
Sure!